subsys/net/l2/virtual/virtual.c

```diff
@@ -212,7 +212,8 @@ static void random_linkaddr(uint8_t *linkaddr, size_t len)
 {
 	sys_rand_get(linkaddr, len);
```

Comment:
Change looks good. The commit message needs some tweaking.
Make it something like:
```
net: virtual: Clear multicast bit when generating link address

The least significant bit of the first octet of a MAC address is a unicast/multicast bit.
The bit should be cleared when generating a random link address for a virtual interface.
Any frames from such interface/address will be dropped by the network as invalid if
the multicast bit is set.

Signed-off-by: <your mail>
```